crates/kubelet-derive/src/lib.rs

```diff
@@ -0,0 +1,89 @@
+//! A crate for deriving state machine traits in Kubelet. Right now this crate only consists of a
+//! derive macro for the `TransitionTo` trait. In addition to the `derive` attribute, this macro
+//! also requires the use of a custom attribute called `transition_to` that specifies the types that
+//! can be transitioned to. Not specifying this attribute will result in a compile time error. A
+//! simple example of this is below:
+//!
+//! ```rust,no_run
+//! use kubelet_derive::TransitionTo;
+//!
+//! pub struct VolumeMount;
+//! pub struct ImagePullBackoff;
+//!
+//! #[derive(Default, Debug, TransitionTo)]
+//! #[transition_to(VolumeMount, ImagePullBackoff)]
+//! pub struct ImagePull;
+//!```
+
+extern crate proc_macro;
+
+use crate::proc_macro::TokenStream;
+use quote::quote;
+use syn::{parse_macro_input, Attribute, DeriveInput, Error, Generics, Ident, Meta, MetaList};
+
+const ATTRIBUTE_NAME: &str = "transition_to";
+
+#[proc_macro_derive(TransitionTo, attributes(transition_to))]
+pub fn derive_transition_to(input: TokenStream) -> TokenStream {
+    let input = parse_macro_input!(input as DeriveInput);
+
+    let name = input.ident;
+    let generics = input.generics;
+
+    let mut transitions = get_transitions(input.attrs);
```

Comment:
No need for this to be mutable here I think?

Reply:
It does so I can pop the data out of the vector down the line